# ReferencesTeam Smith referenced the following technical papers and blogs.Akbar, R., Bashour, H., Rawat, P., Robert, P. A., Smorodina, E., Cotet, T.-S., Flem-Karlsen, K., . . .  & Greiff, V. (2022). Progress and challenges for the machine learning-based design of fit-for-purpose monoclonal antibodies, mAbs, 14(1), 2008790. doi: 10.1080/19420862.2021.2008790Chinery, L., Wahome, N., Moal, I., & Deane, C. M. (2022). Paragraph - antibody paratope prediction using graph neural networks with minimal feature vectors [preprint]. bioRxiv. http://biorxiv.org/lookup/doi/10.1101/2022.06.10.495640 Chungyoun, M., Ruffolo, J., & Gray, J. (2024). FLAb: Benchmarking deep learning methods for antibody fitness prediction [preprint]. bioRxiv. http://biorxiv.org/lookup/doi/10.1101/2024.01.13.575504Cui, A., Di Niro, R., Vander Heiden, J. A., Briggs, A. W., Adams, K., Gilbert, T., O’Connor, K. C., . . . & Kleinstein, S. H. (2016). A model of somatic hypermutation targeting in mice based on high-throughput immunoglobulin sequencing data. J Immunol, 197(9): 3566–3574. doi:10.4049/jimmunol.1502263 Egaji, O. A., Ballard-Smith, S., Asghar, I., & Griffiths, M. (2020). A machine learning approach for predicting antibody properties. ICICM '20: Proceedings of the 10th International Conference on Information Communication and Management. https://dl.acm.org/doi/proceedings/10.1145/3418981Engelhart, E., Emerson, R., Shing , L., Lennartz,C., Guion, D., Kelley, M., Lin, C., Lopez, R., Younger, D., & Walsh, M. E. (2022). A dataset comprised of binding interactions for 104,972 antibodies against a SARS-CoV-2 peptide. Scientific Data, 9(653). https://doi.org/10.1038/s41597-022-01779-4 Graves, J., Byerly, J., Priego, E. Makkapati, N., Parish, S. V., Medellin, B. & Berrondo, M. (2020). Antibodies, 9(12). doi:10.3390/antib9020012 Katwala, A. (2023, Aug. 9). AI is building highly effective antibodies that humans can’t even imagine. Wired UK.Khuat, T. T., Bassett, R., Otte, E., Grevis-James, A., & Gabrys, B. (2024). Applications of machine learning in antibody discovery, process development, manufacturing and formulation: Current trends, challenges, and opportunities. Computers and Chemical Engineering, 182. https://doi.org/10.1016/j.compchemeng.2024.108585 Kim, J., McFee, M., Fang, Q., Abdin, O., & Kim, P. M. (2022). Computational and artificial intelligence-based methods for antibody development. Trends in Pharmacological Sciences, 44(3), 175-189. https://doi.org/10.1016/j.tips.2022.12.005 Li, L., Gupta, E., Spaeth, J., Shing, L., Jaimes, R., Engelhart, E., Lopez, R., . . . & Walsh, M. E. (2023). Machine learning optimization of candidate antibody yields highly diverse sub-nanomolar affinity antibody libraries. Nature Communications, 14. https://doi.org/10.1038/s41467-023-39022-2 Liberis, E., Petar Velickovic, P., Sormanni, P., Vendruscolo, M., & Lio, P. (2018). Parapred: antibody paratope prediction using convolutional and recurrent neural networks. Bioinformatics, 34(17), 2018, 2944–2950. doi: 10.1093/bioinformatics/bty305 Olsen, T. H., Moal, I. H., & Deane, C. M. (2022). AbLang: An antibody language model for completing antibody sequences [preprint]. bioRxiv. doi: https://doi.org/10.1101/2022.01.20.477061 Olsen, T. H., Moal, I. H., & Deane, C. M. (2024). Addressing the antibody germline bias and its effect on language models for improved antibody design [preprint]. bioRxiv. doi: https://doi.org/10.1101/2024.02.02.578678 Park, M., Seo, S.-W., Park, E., & Kim, J. (2022). EpiBERTope: a sequence-based pre-trained BERT model improves linear and structural epitope prediction by learning long-distance protein interactions effectively [preprint]. bioRxiv. doi: https://doi.org/10.1101/2022.02.27.481241 Parkinson, J. & Wang, W. (2024). For antibody sequence generative modeling, mixture models may be all you need [preprint]. bioRxiv. https://doi.org/10.1101/2024.01.27.577555 Ruffolo, J. A., Lee-Shin, C., Mahajon, S. P., & Gray, J. J. (2023). Fast, accurate antibody structure prediction from deep learning on massive set of natural antibodies. Nature Communications, 14. https://doi.org/10.1038/s41467-023-38063-x Schoeder, C. T., Schmitz, S., Adolf-Bryfogle, J., Sevy, A. M., Finn, J. A., Sauer, M. F., Bozhanova, N. G., . . ., & Moretti, R. (2021). Modeling immunity with Rosetta: methods for antibody and antigen design. Biochemistry, 60, 825?846. https://dx.doi.org/10.1021/acs.biochem.0c00912 Shanehsazzadeh, A.,  Alverio, J., Kasun, G., Levine, S., Khan, J. A., Chung, C., & Diaz, N. In vitro validated antibody design against multiple therapeutic antigens using generative inverse folding [preprint]. bioRxiv. doi: https://doi.org/10.1101/2023.12.08.570889 Smith, T. (2018, Nov. 7). What is Immunoprofiling? Blog. Digital World Biology. Accessed 27 Feb. 2024. https://digitalworldbiology.com/blog/what-immunoprofiling Wu, J., Wu, F., Jiang, B., Liu, W., & Zhao. P. (2022). tFold-Ab: Fast and accurate antibody structure prediction without sequence homologs. Machine Learning for Structural Biology Workshop, NeurIPS.Yang, Y. X., Huang, J. H., & Zhu, B. T. (2023). AREA-AFFINITY: A Web Server for Machine Learning-Based Prediction of Protein?Protein and Antibody?Protein Antigen Binding Affinities. Journal of Chemical Information and Modeling. 63, 3230-3237. https://doi.org/10.1021/acs.jcim.2c01499 